### PR TITLE
Log Watson events in AppInsights

### PR DESCRIPTION
**Pull request by amcasey**
Affects the C# REPL, the VB REPL, and the IDE.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @heejaechang @mavasani for telemetry
FYI @tmat @cston @KevinH-MS @genlu @ManishJayaswal  for Interactive
FYI @DustinCampbell @Pilchie @jasonmalinowski @CyrusNajmabadi  for IDE
FYI @dpoeschl because I was bugging him about the linked files involved in this change

**Comment by tmat** [comment]
:+1:

**Comment by heejaechang** [comment]
:+1: 

@amcasey as you might know, the watson code is from debugger team, so it might be good idea to include him (peter?) to see whether they have changed any code around watson.

**Comment by amcasey** [comment]
@heejaechang I started with the debugger code because they were already reporting telemetry.  The only interesting update was suppressing window opening.

File: src/VisualStudio/InteractiveServices/Interactive/VsInteractiveWindowPackage.cs [hunk 1]
```diff
@@ -34,10 +37,17 @@ protected override void Initialize()
...
 
             // Load the Roslyn package so that its FatalError handlers are hooked up.
-            IVsPackage setupPackage;
+            IVsPackage roslynPackage;
             var shell = (IVsShell)this.GetService(typeof(SVsShell));
-            shell.LoadPackage(Guids.RoslynPackageId, out setupPackage);
-            shell.LoadPackage(Microsoft.VisualStudio.InteractiveWindow.Shell.Guids.InteractiveWindowPackageId, out setupPackage);
+            shell.LoadPackage(Guids.RoslynPackageId, out roslynPackage);
+            Debug.Assert(core::Microsoft.CodeAnalysis.ErrorReporting.FatalError.Handler != null);
+            Debug.Assert(core::Microsoft.CodeAnalysis.ErrorReporting.FatalError.NonFatalHandler != null);
```
Comment (amcasey) [inline src/VisualStudio/InteractiveServices/Interactive/VsInteractiveWindowPackage.cs]: Motivating example for https://github.com/dotnet/roslyn/issues/5349.

**Comment by amcasey** [comment]
Merging slightly prematurely so that we'll get telemetry from signed builds ASAP.  Please feel free to leave additional comments.

**Pull request merged**